mistralrs-core/src/utils/model_config.rs

```diff
@@ -0,0 +1,328 @@
+        ct: gguf_file::Content,
+        reader: &mut R,
+        device: &candle_core::Device,
+        lora_config: &[((String, String), LoraConfig)],
+        vb: &VarBuilder,
+        ordering: &Ordering,
+        xlora_config: Option<XLoraConfig>,
+        mapper: DeviceMapMetadata,
+        preload_adapters: &Option<HashMap<String, (VarBuilder, LoraConfig)>>,
+    ) -> Result<Self, candle_core::Error> where Self: Sized;
+}
+
+// NOTE: Below is a workaround to proxy params to the existing API methods `get_gguf()` / `get_gmml()` traits covered above.
+impl Config<GGML, NoAdapter> {
+    pub fn try_into_model<T: FromGGML>(self) -> Result<T, candle_core::Error> {
+        // Destructure props:
+        let GGML(
+            FileGGML { ct, gqa },
+        ) = self.quant;
+
+        // Forwards all structured fields above into the required flattened param sequence:
+        T::from_ggml(
+            ct,
+            gqa,
+        )
+    }
+}
+
+impl Config<GGML, Adapter<'_>> {
+    pub fn try_into_model<T: FromAdapterGGML>(self) -> Result<T, candle_core::Error> {
+        // Destructure props:
+        let GGML(
+            FileGGML { ct, gqa },
+        ) = self.quant;
+
+        let Adapter {
+            xlora_config,
+            lora_config,
+            vb,
+            ordering,
+            preload_adapters,
+        } = self.adapter;
+
+        // Forwards all structured fields above into the required flattened param sequence:
+        T::from_ggml(
+            ct,
+            gqa,
+            lora_config,
+            &vb,
+            ordering,
+            xlora_config,
+            &preload_adapters,
+        )
+    }
+}
+
+impl Config<GGUF<'_>, NoAdapter> {
+    pub fn try_into_model<T: FromGGUF>(self) -> Result<T, candle_core::Error> {
+        // Destructure props:
+        let GGUF(
+            FileGGUF { ct, reader },
+            Device { device, mapper },
+        ) = self.quant;
+
+        // Forwards all structured fields above into the required flattened param sequence:
+        T::from_gguf(
+            ct,
+            reader,
+            device,
+            mapper,
+        )
+    }
+}
+
+impl Config<GGUF<'_>, Adapter<'_>> {
+    pub fn try_into_model<T: FromAdapterGGUF>(self) -> Result<T, candle_core::Error> {
+        // Destructure props:
+        let GGUF(
+            FileGGUF { ct, reader },
+            Device { device, mapper },
+        ) = self.quant;
+
+        let Adapter {
+            xlora_config,
+            lora_config,
+            vb,
+            ordering,
+            preload_adapters,
+        } = self.adapter;
+
+        // Forwards all structured fields above into the required flattened param sequence:
+        T::from_gguf(
+            ct,
+            reader,
+            device,
+            lora_config,
+            &vb,
+            ordering,
+            xlora_config,
+            mapper,
+            &preload_adapters,
+        )
+    }
+}

```

Comment:
All this is just mapping to those traits expected sequence of params. It'll disappear too once those models `from_gguf()` / `from_ggml()` take the param structs instead.

I don't think there's much value in calling those directly (_if that's supported by the Rust / Python APIs_) vs the loaders, so I assume the more logical grouping with structs for fewer param inputs (and less strict order) will be an improvement 👍